app/src/main/java/org/traccar/client/StatusWidget.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016 - 2022 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.client
+
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.*
+import android.widget.RemoteViews
+
+import androidx.preference.PreferenceManager
+
+/**
+ * Implementation of Status Widget functionality, which can display the status of the service -
+ * whether it is enabled or not.
+ */
+class StatusWidget : AppWidgetProvider() {
+
+    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
+        update(context, appWidgetManager, appWidgetIds)
+    }
+
+    override fun onEnabled(context: Context) {
+        // When the first widget is created, we want to start receiving service status broadcasts,
+        // so we can react and change the look
+        val filter = IntentFilter()
+        filter.addAction(TrackingService.ACTION_STARTED)
+        filter.addAction(TrackingService.ACTION_STOPPED)
+        // Note: we must register through the app context as a workaround
+        // for 'BroadcastReceiver components are not allowed to register to receive intents'
+        context.applicationContext.registerReceiver(Companion, filter)
```

Comment:
Does it have to be `applicationContext`?

Reply:
I see no other options.
According to the comment this trick was needed because of the framework restrictions.

Reply:
What is the restriction? Do you have a link to the source?

Reply:
I couldn't find the official source for this restriction, and the trick was found somewhere on stackoverflow ([example](https://stackoverflow.com/a/37788987/536172)).
If you try to register the receiver as ` context.registerReceiver(Companion, filter)`, the exception will be raised